src/shogun/mathematics/linalg/linalg.h

```diff
@@ -35,40 +35,40 @@
 #include <shogun/lib/config.h>
 
 /**
- * Just include this file to use in your applications.
+ * Just include this file to use in your applications (in the cpp)
  */
 
-/**
- * This namespace contains all linear algebra specific modules and operations.
- * The rest of the library is available only when the one of the supported
- * backend exists in the local system
- */
-
-#ifdef HAVE_LINALG_LIB
 namespace shogun
 {
 
+/**
+ * This namespace contains all linear algebra specific modules and operations
+ * for which we (may) rely on multiple implementation, either native or
+ * making use of some supported third party external linear algebra libraries.
+ */
 namespace linalg
 {
 
 /**
  * Developer's Note :
- * - Changing the default backend would just require to change it in the following
- *   enum
+ * - Changing the default backend would just require to change it in the
+ *   Backend enum
  * - Please use the same names as HAVE_<BACKEND> macros (e.g. HAVE_EIGEN3)
+ * - Please refer to the developer's wiki (link) for the design and structure
+ *   of internal linalg module
  */
 
 /**
  * @brief
  * All currently supported linear algebra backend libraries, with a default
  * backend, which will be used for all the tasks if any particular backend is
- * not set explicitly via cmake options.
+ * not set explicitly via cmake options. A Native backend is also defined to
+ * accommodate Shogun's native implementation of some operations.
  *
  * The enum defines these backends in order of priority as default backend, as
- * in, first defined one will be used as default
- *
- * Note - Currently EIGEN3 is the default (if its available)
+ * in, first defined one will be used as default.
  *
+ * Note - Currently EIGEN3 is the default (if it is available).
```

Comment:
`DEFAULT` will be same as the first thing defined. So if both Eigen3 and ViennaCL are there, it is Eigen3. If only ViennaCL is there it is ViennaCL. If nothing is there it is NATIVE (which is limited to only a few operations that we'll provide native implementation).